From gregkh@suse.de Thu May 15 21:04:51 2008
X-Envelope-From: <stable-commits-owner@vger.kernel.org>
X-Envelope-To: <vger@sl-h.de>
X-Delivery-Time: 1210881085
X-UID: 237682
Return-Path: <stable-commits-owner@vger.kernel.org>
X-RZG-CLASS-ID: mi
Received: from vger.kernel.org ([209.132.176.167])
	by mailin.webmailer.de (christine mi55) (RZmta 16.35)
	with ESMTP id T055a0k4FJpKSo for <vger@sl-h.de>;
	Thu, 15 May 2008 21:51:25 +0200 (MEST)
	(envelope-from: <stable-commits-owner@vger.kernel.org>)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1760248AbYEOTut (ORCPT <rfc822;vger@sl-h.de>);
	Thu, 15 May 2008 15:50:49 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id S1760125AbYEOTus
	(ORCPT <rfc822;stable-commits-outgoing>);
	Thu, 15 May 2008 15:50:48 -0400
Received: from mx2.suse.de ([195.135.220.15]:54407 "EHLO mx2.suse.de"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
	id S1759652AbYEOTuq (ORCPT <rfc822;stable-commits@vger.kernel.org>);
	Thu, 15 May 2008 15:50:46 -0400
Received: from Relay1.suse.de (relay-ext.suse.de [195.135.221.8])
	(using TLSv1 with cipher DHE-RSA-AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mx2.suse.de (Postfix) with ESMTP id 3CFC7457B4;
	Thu, 15 May 2008 21:50:45 +0200 (CEST)
Subject: patch usb-add-zoom-telephonics-model-3095f-v.92-usb-mini-external-modem-to-cdc-acm.patch added to 2.6.25-stable tree
To: iain@imcfarla.homelinux.net,
 gregkh@suse.de,
 oneukum@suse.de
Cc: <stable@kernel.org>,
 <stable-commits@vger.kernel.org>
From:	<gregkh@suse.de>
Date:	Thu, 15 May 2008 12:04:51 -0700
In-Reply-To: <200805141921.m4EJLJ6Q019979@hera.kernel.org>
Message-Id: <20080515195044.3C6A3141BC25@imap.suse.de>
Sender:	stable-commits-owner@vger.kernel.org
Precedence: bulk
Reply-To: linux-kernel@vger.kernel.org
List-ID: <stable-commits.vger.kernel.org>
X-Mailing-List:	stable-commits@vger.kernel.org
Status: R
X-Status: NC
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  


This is a note to let you know that we have just queued up the patch titled

     Subject: USB: add Zoom Telephonics Model 3095F V.92 USB Mini External modem to cdc-acm

to the 2.6.25-stable tree.  Its filename is

     usb-add-zoom-telephonics-model-3095f-v.92-usb-mini-external-modem-to-cdc-acm.patch

A git repo of this tree can be found at 
    http://www.kernel.org/git/?p=linux/kernel/git/stable/stable-queue.git;a=summary


>From stable-bounces@linux.kernel.org Wed May 14 12:21:34 2008
From: Iain McFarlane <iain@imcfarla.homelinux.net>
Date: Wed, 14 May 2008 19:21:19 GMT
Subject: USB: add Zoom Telephonics Model 3095F V.92 USB Mini External modem to cdc-acm
To: jejb@kernel.org, stable@kernel.org
Message-ID: <200805141921.m4EJLJ6Q019979@hera.kernel.org>

From: Iain McFarlane <iain@imcfarla.homelinux.net>

commit 6149ed5e3a6207595bd7362af7724d64f44af216 upstream

The patch below is a necessary workaround to support the Zoom Telephonics Model 3095F V.92 USB Mini External modem, which fails to initialise properly during normal probing thus:

May  3 22:53:00 imcfarla kernel: drivers/usb/class/cdc-acm.c: Zero length descriptor references
May  3 22:53:00 imcfarla kernel: cdc_acm: probe of 5-2:1.0 failed with error -22

Adding the patch below causes the probing section to be skipped, and the modem
then initialises correctly.

Signed-off-by: Iain McFarlane <iain@imcfarla.homelinux.net>
Acked-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-acm.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -1242,6 +1242,9 @@ static struct usb_device_id acm_ids[] = 
 	{ USB_DEVICE(0x22b8, 0x7000), /* Motorola Q Phone */
 	.driver_info = NO_UNION_NORMAL, /* has no union descriptor */
 	},
+	{ USB_DEVICE(0x0803, 0x3095), /* Zoom Telephonics Model 3095F USB MODEM */
+	.driver_info = NO_UNION_NORMAL, /* has no union descriptor */
+	},
 
 	/* control interfaces with various AT-command sets */
 	{ USB_INTERFACE_INFO(USB_CLASS_COMM, USB_CDC_SUBCLASS_ACM,


Patches currently in stable-queue which might be from iain@imcfarla.homelinux.net are

queue-2.6.25/usb-add-zoom-telephonics-model-3095f-v.92-usb-mini-external-modem-to-cdc-acm.patch
--
To unsubscribe from this list: send the line "unsubscribe stable-commits" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

